packages/renderer/src/lib/preferences/PreferencesConnectionDetailsTerminal.spec.ts

```diff
@@ -0,0 +1,216 @@
+      socketPath: '/socket/path',
+    },
+  } as unknown as ProviderContainerConnectionInfo;
+
+  let onDataCallback: (data: string) => void = () => {};
+
+  const sendCallbackId = 12345;
+  shellInProviderConnectionMock.mockImplementation(
+    (
+      _providerId: string,
+      _providerConnectionInfo: ProviderContainerConnectionInfo | ProviderKubernetesConnectionInfo,
+      onData: (data: string) => void,
+      _onError: (error: string) => void,
+      _onEnd: () => void,
+    ) => {
+      onDataCallback = onData;
+      // return a callback id
+      return Promise.resolve(sendCallbackId);
+    },
+  );
+
+  // render the component with a terminal
+  let renderObject = render(PreferencesConnectionDetailsTerminal, { provider, connectionInfo, screenReaderMode: true });
+
+  // wait shellInProviderMock is called
+  await waitFor(() => expect(shellInProviderConnectionMock).toHaveBeenCalled());
+
+  // write some data on the terminal
+  onDataCallback('hello\nworld');
+
+  // wait 1s
+  await new Promise(resolve => setTimeout(resolve, 1000));
+
+  // search a div having aria-live="assertive" attribute
+  const terminalLinesLiveRegion = renderObject.container.querySelector('div[aria-live="assertive"]');
+
+  // check the content
+  expect(terminalLinesLiveRegion).toHaveTextContent('hello world');
+
+  // should be no terminal being stored
+  const terminals = get(providerTerminals);
+  expect(terminals.length).toBe(0);
+
+  // destroy the object
+  renderObject.unmount();
+
+  // now, check that we have a terminal that is in the store
+  const terminalsAfterDestroy = get(providerTerminals);
+  expect(terminalsAfterDestroy.length).toBe(1);
+
+  // ok, now render a new terminal widget, it should reuse data from the store
+  renderObject = render(PreferencesConnectionDetailsTerminal, { provider, connectionInfo, screenReaderMode: true });
+
+  // wait shellInProviderMock is called
+  await waitFor(() => expect(shellInProviderConnectionMock).toHaveBeenCalledTimes(2));
+
+  // wait 1s that everything is done
+  await new Promise(resolve => setTimeout(resolve, 1000));
+
+  const terminalLinesLiveRegion2 = renderObject.container.querySelector('div[aria-live="assertive"]');
+
+  // check the content
+  expect(terminalLinesLiveRegion2).toHaveTextContent('hello world');
+
+  // creating a new terminal requires new shellInProvider call
+  expect(shellInProviderConnectionMock).toHaveBeenCalledTimes(2);
+});
+
+test('terminal active/ restarts connection after stopping and starting a provider', async () => {
+  const provider: ProviderInfo = {
+    id: 'myProvider',
+    internalId: 'myInternalProvider',
+    status: 'started',
+  } as unknown as ProviderInfo;
+
+  const connectionInfo: ProviderContainerConnectionInfo = {
+    name: 'myConnection',
+    status: 'started',
+    endpoint: {
+      socketPath: '/socket/path',
+    },
+  } as unknown as ProviderContainerConnectionInfo;
+
+  let onDataCallback: (data: string) => void = () => {};
```

Comment:
```suggestion
  let onDataCallback: ((data: string) => void) | undefined
```

then we you need to ensure it is defined you use `expect(onDataCallback).toBeDefined()` then when you need to call it `onDataCallback?.()`